src/ClientHandle.cpp

```diff
@@ -1529,34 +1532,42 @@ void cClientHandle::HandleRightClick(int a_BlockX, int a_BlockY, int a_BlockZ, e
 
 void cClientHandle::HandleChat(const AString & a_Message)
 {
-	// We no longer need to postpone message processing, because the messages already arrive in the Tick thread
-
-	// If a command, perform it:
-	AString Message(a_Message);
-	if (cRoot::Get()->GetServer()->Command(*this, Message))
-	{
-		return;
-	}
-
-	// Not a command, broadcast as a message:
-	cCompositeChat Msg;
-	AString Color = m_Player->GetColor();
-	if (Color.length() == 3)
+	if ((a_Message.size()) > MAX_CHAT_MSG_LENGTH)
 	{
-		Color = AString("@") + Color[2];
+		this->Kick(std::string("Please don't exceed the maximum message length of ")
+		+ std::to_string(MAX_CHAT_MSG_LENGTH));
 	}
```

Comment:
Please write this as a guard clause to avoid deep nesting of `if`-`else` blocks:
```cpp
void cClientHandle::HandleChat(const AString & a_Message)
{
    if (a_Message.size() > MAX_CHAT_MSG_LENGTH) {
        Kick(std::string("Please don't exceed the maximum message length of ")
	    	+ std::to_string(MAX_CHAT_MSG_LENGTH));
        return;
    }

    // Rest of code exactly as before
}
```